SPECS/kubernetes/CVE-2024-45310.patch

```diff
@@ -0,0 +1,673 @@
++			if err != nil {
++				return "", fmt.Errorf("make parent dir of file bind-mount: %w", err)
++			}
++			defer destDirFd.Close()
++			// Make the target file. We want to avoid opening any file that is
++			// already there because it could be a "bad" file like an invalid
++			// device or hung tty that might cause a DoS, so we use mknodat.
++			// destBase does not contain any "/" components, and mknodat does
++			// not follow trailing symlinks, so we can safely just call mknodat
++			// here.
++			if err := unix.Mknodat(int(destDirFd.Fd()), destBase, unix.S_IFREG|0o644, 0); err != nil {
++				// If we get EEXIST, there was already an inode there and
++				// we can consider that a success.
++				if !errors.Is(err, unix.EEXIST) {
++					err = &os.PathError{Op: "mknod regular file", Path: dest, Err: err}
++					return "", fmt.Errorf("create target of file bind-mount: %w", err)
++				}
++			}
++			// Nothing left to do.
++			return dest, nil
++		}
++
++	case "tmpfs":
++		// If the original target exists, copy the mode for the tmpfs mount.
++		if stat, err := os.Stat(dest); err == nil {
++			dt := fmt.Sprintf("mode=%04o", syscallMode(stat.Mode()))
++			if m.Data != "" {
++				dt = dt + "," + m.Data
++			}
++			m.Data = dt
++
++			// Nothing left to do.
++			return dest, nil
++		}
++	}
++
++	if err := utils.MkdirAllInRoot(rootfs, dest, 0o755); err != nil {
++		return "", err
++	}
++	return dest, nil
++}
++
+ func mountToRootfs(m *configs.Mount, c *mountConfig) error {
+ 	rootfs := c.root
+ 
+@@ -435,57 +474,34 @@ func mountToRootfs(m *configs.Mount, c *mountConfig) error {
+ 		} else if !fi.IsDir() {
+ 			return fmt.Errorf("filesystem %q must be mounted on ordinary directory", m.Device)
+ 		}
+-		if err := os.MkdirAll(dest, 0o755); err != nil {
++		if err := utils.MkdirAllInRoot(rootfs, dest, 0o755); err != nil {
+ 			return err
+ 		}
+ 		// Selinux kernels do not support labeling of /proc or /sys.
+ 		return mountPropagate(m, rootfs, "", nil)
+ 	}
+ 
+-	mountLabel := c.label
+ 	mountFd := c.fd
+-	dest, err := securejoin.SecureJoin(rootfs, m.Destination)
++	dest, err := createMountpoint(rootfs, m, mountFd, m.Source)
+ 	if err != nil {
+-		return err
+-	}
++		return fmt.Errorf("create mount destination for %s mount: %w", m.Destination, err)
++ 	}
++	mountLabel := c.label
+ 
+ 	switch m.Device {
+ 	case "mqueue":
+-		if err := os.MkdirAll(dest, 0o755); err != nil {
+-			return err
+-		}
+ 		if err := mountPropagate(m, rootfs, "", nil); err != nil {
+ 			return err
+ 		}
+ 		return label.SetFileLabel(dest, mountLabel)
+ 	case "tmpfs":
+-		stat, err := os.Stat(dest)
+-		if err != nil {
+-			if err := os.MkdirAll(dest, 0o755); err != nil {
+-				return err
+-			}
+-		}
+-
+ 		if m.Extensions&configs.EXT_COPYUP == configs.EXT_COPYUP {
+ 			err = doTmpfsCopyUp(m, rootfs, mountLabel)
+ 		} else {
+ 			err = mountPropagate(m, rootfs, mountLabel, nil)
+ 		}
+-
+-		if err != nil {
+-			return err
+-		}
+-
+-		if stat != nil {
```

Comment:
As per https://github.com/opencontainers/runc/security/advisories/GHSA-jfvp-7x6p-h2pv for Release-1.1, I don't see this change in any of the 2 patches:
https://github.com/opencontainers/runc/commit/8781993968fd964ac723ff5f360b6f259e809a3e, or
https://github.com/opencontainers/runc/commit/f0b652ea61ff6750a8fcc69865d45a7abf37accf#diff-e0932af06cb46065b2ef5fc89bc2bd9b880c9578e01b89ef4ecaeffb78a8bef0

Please explain this.